ui/appui-react/src/appui-react/childwindow/ChildWindowManager.tsx

```diff
@@ -7,207 +7,131 @@
  * @module ChildWindowManager
  */
 
-import "./ChildWindowManager.scss";
 import * as React from "react";
-import * as ReactDOM from "react-dom";
-import { copyStyles } from "./CopyStyles";
-import { Provider } from "react-redux";
-import { StateManager } from "../redux/StateManager";
-import { UiStateStorageHandler } from "../uistate/useUiStateStorage";
-import { PopupRenderer } from "../popup/PopupManager";
-import { ModelessDialogRenderer } from "../dialog/ModelessDialogManager";
-import { ModalDialogRenderer } from "../dialog/ModalDialogManager";
-import { CursorPopupMenu } from "../cursor/cursormenu/CursorMenu";
-import { FrameworkVersion } from "../hooks/useFrameworkVersion";
-import { FrontstageManager } from "../frontstage/FrontstageManager";
-import { ThemeManager } from "../theme/ThemeManager";
-
-const childHtml = `<!DOCTYPE html>
-<html>
-<head>
-  <meta charset="utf-8" />
-  <style>
-    html,
-    body {
-      height: 100%;
-      width: 100%;
-      margin: 0;
-      overflow: hidden;
-    }
-    #root {
-      height: 100%;
-    }
-  </style>
-</head>
-<body>
-  <noscript>You need to enable JavaScript to run this app.</noscript>
-  <div id="root"></div>
-</body>
-</html>`;
-
-/** @public */
-export interface OpenChildWindowInfo {
-  childWindowId: string;
-  window: Window;
-  parentWindow: Window;
-}
-
-/** @public */
-export interface ChildWindowLocationProps {
-  width: number;
-  height: number;
-  left: number;
-  top: number;
-}
+import { ChildWindowLocationProps, OpenChildWindowInfo } from "../framework/FrameworkChildWindows";
+import { InternalChildWindowManager } from "./InternalChildWindowManager";
 
 /** Supports opening a child browser window from the main application window. The child window is managed by the main application
  * and is running in the same security context. The application must deliver the html file iTwinPopup.html along side its index.html.
  * See also: [Child Window Manager]($docs/learning/ui/appui-react/ChildWindows.md)
- * @public */
+ * @public
+ * @deprecated in 3.6. Use `UiFramework.childWindows` property to access.
+ * */
 export class ChildWindowManager {
```

Comment:
Since the new API is beta, and this class is public, extract-api sees a conflict so I cant mention that it implement the interface. All others are implemented by static classes so the "implements" clause do not work. We should adress this however when we make the switch to instances implementation (which I think should be the next step in this process)